ocaml/typing/typetexp.ml

```diff
@@ -976,6 +953,38 @@ and transl_type_aux env ~row_context ~aliased ~policy mode styp =
             pack_fields = ptys;
             pack_txt = p;
            }) ty
+  | Ptyp_functor (arg_label, name, (p, l), st) ->
+    let path, mty, ptys =
+      transl_package env ~policy ~row_context styp.ptyp_loc p l in
+    let scoped_ident, cty =
+      with_local_level begin fun () ->
+        let scoped_ident =
+          Ident.create_scoped ~scope:(Ctype.get_current_level()) name.txt
+        in
+        let env = Env.add_module scoped_ident Mp_present mty env in
+        scoped_ident,
+          transl_type env ~policy ~row_context (get_alloc_mode styp) st
+      end in
+    let ident = Ident.create_unscoped name.txt in
+    let ctyp_type =
+        Option.value ~default:cty.ctyp_type
+          (instance_funct ~id_in:scoped_ident ~p_out:(Pident (Ident.of_unscoped ident))
+                          ~fixed:false cty.ctyp_type)
```

Comment:
The problem is that we can never instantiate a unification variable by type constructor containing an unscoped identifier.

For exemple it would pose a problem with the code below when unifiying the return type `M.t` with the type of `x ` at that moment `'a` if `M` was an unscoped identifier.
```ocaml
let f (module M : T) x : M.t = x
```
Is there a moment where a type is being built directly with an unscoped identifier ? It could be possible to use unscoped identifiers directly if the know that we won't use the levels inside that identifier before closing the type with the binder